exosphere-shared/src/docker-helper.ls

```diff
@@ -1,146 +1,49 @@
+      | err => done err
+      DockerHelper._force-remove-containers containers, done
 
-  @run-all-images = ({env, write}, cb) ->
-    new ObservableProcess('docker-compose up'
-                          env: env
-                          stdout: {write}
-                          stderr: {write})
-      ..on 'ended', cb
 
+  @remove-all-containers = (done) ->
+    docker.list-containers (err, containers) ->
+      | err => done err
+      DockerHelper._force-remove-containers containers, done
 
-  @kill-container = ({service-name, write}, cb) ->
-    new ObservableProcess("docker-compose kill #{service-name}"
-                          stdout: {write}
-                          stderr: {write})
-      ..on 'ended', cb
 
+  @_force-remove-containers = (containers, done) ->
+    for container in containers
+      docker.get-container(container.Id).remove {force:true}, (err) ->
+        | err => done err
+        done!
 
-  @kill-all-containers = ({write}, cb) ->
-    new ObservableProcess('docker-compose down'
-                          stdout: {write}
-                          stderr: {write})
-      ..on 'ended', cb
 
+  @_container-is-running = (name, done) ->
+    docker.list-containers {name}, (err, containers) ->
+      | err => done err
+      done null, any (.Names.includes name), containers
 
-  @create-new-container = ({service-name, env, write}, cb) ->
-    new ObservableProcess("docker-compose create --build #{service-name}"
-                          env: env
-                          stdout: {write}
-                          stderr: {write})
-      ..on 'ended', cb
-
-
-  @start-container = ({service-name, env, write}, cb) ->
-    new ObservableProcess("docker-compose restart #{service-name}"
-                          env: env
-                          stdout: {write}
-                          stderr: {write})
-      ..on 'ended', cb
-
-
-  @container-exists = (container) ->
-    child_process.exec-sync('docker ps -a --format {{.Names}}') |> (.to-string!) |> (.split '\n') |> (.includes container)
-
-
-  @container-is-running = (container-name) ->
-    child_process.exec-sync('docker ps --format {{.Names}}/{{.Status}}') |> (.to-string!) |> (.split os.EOL) |> any (.includes "#{container-name}/Up")
-
-
-  @ensure-container-is-running = (container, done) ~>
-    | DockerHelper.container-is-running container.container-name  =>  return done!
-    | DockerHelper.container-exists container.container-name      =>  DockerHelper.start-container container, done
-    | otherwise                                                   =>  DockerHelper.run-image container, done
-
-
-  @get-build-command = (image, build-flags) ->
-    return "docker build -t #{image.author}/#{image.name} #{if build-flags then build-flags else ""} ."
-
-
-  @get-config = (image) ->
-    child_process.exec-sync("docker run --rm=true #{image} cat service.yml", 'utf8') |> (.to-string!)
-
-
-  @get-docker-ip = (container) ->
```

Comment:
No, there's no need to manually query container IPs in order to link them to each other. `docker-compose` handles linking between containers now.